// ngx-communicate/esm2015/lib/ngx-communicate.service.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/ngx-communicate.service.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Injectable } from "@angular/core";
import * as i0 from "@angular/core";
export class NgxCommunicateService {
    constructor() {
        this._map = new Map();
    }
    /**
     * \@method send
     * \@property data
     * @param {?} event
     * @param {?} data
     * @return {?} void
     */
    broadcast(event, data) {
        /** @type {?} */
        const subscribers = this._map.get(event) || [];
        subscribers.forEach((/**
         * @param {?} subscriber
         * @return {?}
         */
        subscriber => subscriber(data)));
    }
    /**
     * \@method on
     * \@description Subscribes to an event
     * \@var event Event Name / callback the function that will return the data of that event
     * @param {?} event
     * @param {?} callback
     * @return {?}
     */
    on(event, callback) {
        /** @type {?} */
        let subscribers = this._map.get(event);
        if (subscribers == undefined)
            subscribers = new Array();
        subscribers.push(callback);
        this._map.set(event, subscribers);
    }
}
NgxCommunicateService.decorators = [
    { type: Injectable, args: [{
                providedIn: "root"
            },] }
];
/** @nocollapse */
NgxCommunicateService.ctorParameters = () => [];
/** @nocollapse */ NgxCommunicateService.ngInjectableDef = i0.ɵɵdefineInjectable({ factory: function NgxCommunicateService_Factory() { return new NgxCommunicateService(); }, token: NgxCommunicateService, providedIn: "root" });
if (false) {
    /**
     * @type {?}
     * @private
     */
    NgxCommunicateService.prototype._map;
}
