// ngx-communicate/esm5/lib/ngx-communicate.service.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/ngx-communicate.service.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Injectable } from "@angular/core";
import * as i0 from "@angular/core";
var NgxCommunicateService = /** @class */ (function () {
    function NgxCommunicateService() {
        this._map = new Map();
    }
    /**
     * @method send
     * @property data
     * @returns void
     */
    /**
     * \@method send
     * \@property data
     * @param {?} event
     * @param {?} data
     * @return {?} void
     */
    NgxCommunicateService.prototype.broadcast = /**
     * \@method send
     * \@property data
     * @param {?} event
     * @param {?} data
     * @return {?} void
     */
    function (event, data) {
        /** @type {?} */
        var subscribers = this._map.get(event) || [];
        subscribers.forEach((/**
         * @param {?} subscriber
         * @return {?}
         */
        function (subscriber) { return subscriber(data); }));
    };
    /**
     * @method on
     * @description Subscribes to an event
     * @var event Event Name
     * @var callback the function that will return the data of that event
     */
    /**
     * \@method on
     * \@description Subscribes to an event
     * \@var event Event Name / callback the function that will return the data of that event
     * @param {?} event
     * @param {?} callback
     * @return {?}
     */
    NgxCommunicateService.prototype.on = /**
     * \@method on
     * \@description Subscribes to an event
     * \@var event Event Name / callback the function that will return the data of that event
     * @param {?} event
     * @param {?} callback
     * @return {?}
     */
    function (event, callback) {
        /** @type {?} */
        var subscribers = this._map.get(event);
        if (subscribers == undefined)
            subscribers = new Array();
        subscribers.push(callback);
        this._map.set(event, subscribers);
    };
    NgxCommunicateService.decorators = [
        { type: Injectable, args: [{
                    providedIn: "root"
                },] }
    ];
    /** @nocollapse */
    NgxCommunicateService.ctorParameters = function () { return []; };
    /** @nocollapse */ NgxCommunicateService.ngInjectableDef = i0.ɵɵdefineInjectable({ factory: function NgxCommunicateService_Factory() { return new NgxCommunicateService(); }, token: NgxCommunicateService, providedIn: "root" });
    return NgxCommunicateService;
}());
export { NgxCommunicateService };
if (false) {
    /**
     * @type {?}
     * @private
     */
    NgxCommunicateService.prototype._map;
}
